src/Tool/Newsletter.php

```diff
@@ -204,7 +204,12 @@ public static function sendNewsletterDocumentBasedMail(Mail $mail, SendingParamC
 
     protected static function obfuscateEmail(string $email): string
     {
-        return substr_replace($email, '.xxx', strrpos($email, '.'));
+        $offset = strrpos($email, '.');
+        if ($offset !== false) {
```

Comment:
```suggestion
        $offset = strrpos($email, '.');
	$offsetAt = strrpos($email, '@');
        if ($offset !== false && $offsetAt !== false && $offset > $offsetAt) {
```

if the email is `name.surname@domain` would still replace to `name.xxx`

Reply:
But the string cannot have an @ character either. So the if should be:
`if ($offset !== false && $offsetAt !== false && $offset > $offsetAt) {`